pkg/webhook/validation/dynakube/oneagent.go

```diff
@@ -106,36 +107,51 @@ func hasConflictingMatchLabels(labelMap, otherLabelMap map[string]string) bool {
 	return labelSelector.Matches(otherLabelSelectorLabels) || otherLabelSelector.Matches(labelSelectorLabels)
 }
 
-func hasOneAgentVolumeStorageEnabled(dynakube *dynatracev1beta2.DynaKube) (isEnabled bool, isSet bool) {
-	envVar := env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentEnableVolumeStorageEnvVarName)
+func hasOneAgentVolumeStorageEnabled(dk *dynakube.DynaKube) (isEnabled bool, isSet bool) {
+	envVar := env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentEnableVolumeStorageEnvVarName)
 	isSet = envVar != nil
 	isEnabled = isSet && envVar.Value == "true"
 
 	return
 }
 
-func unsupportedOneAgentImage(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	if env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentInstallerScriptUrlEnvVarName) != nil ||
-		env.FindEnvVar(dynakube.GetOneAgentEnvironment(), oneagentInstallerTokenEnvVarName) != nil {
+func unsupportedOneAgentImage(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	if env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentInstallerScriptUrlEnvVarName) != nil ||
+		env.FindEnvVar(dk.GetOneAgentEnvironment(), oneagentInstallerTokenEnvVarName) != nil {
 		return warningOneAgentInstallerEnvVars
 	}
 
 	return ""
 }
 
-func conflictingOneAgentVolumeStorageSettings(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	volumeStorageEnabled, volumeStorageSet := hasOneAgentVolumeStorageEnabled(dynakube)
-	if dynakube.NeedsReadOnlyOneAgents() && volumeStorageSet && !volumeStorageEnabled {
+func conflictingOneAgentVolumeStorageSettings(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	volumeStorageEnabled, volumeStorageSet := hasOneAgentVolumeStorageEnabled(dk)
+	if dk.NeedsReadOnlyOneAgents() && volumeStorageSet && !volumeStorageEnabled {
 		return errorVolumeStorageReadOnlyModeConflict
 	}
 
 	return ""
 }
 
-func conflictingHostGroupSettings(_ context.Context, _ *dynakubeValidator, dynakube *dynatracev1beta2.DynaKube) string {
-	if dynakube.HostGroupAsParam() != "" {
+func conflictingHostGroupSettings(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	if dk.HostGroupAsParam() != "" {
 		return warningHostGroupConflict
 	}
 
 	return ""
 }
+
+var oneAgentSemVerVersionSpec = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)$`)
+
+func validateOneAgentVersionIsSemVerCompliant(_ context.Context, _ *dynakubeValidator, dk *dynakube.DynaKube) string {
+	agentVersion := dk.CustomOneAgentVersion()
+	if agentVersion == "" {
+		return ""
+	}
+
+	if !oneAgentSemVerVersionSpec.MatchString(agentVersion) {
```

Comment:
```suggestion
        version := "v"+agentVersion
	if !(semver.IsValid(version) && semver.Prerelease(version) == "" && semver.Build(version) == "" && len(strings.Split(version, ".")) == 3) {
```

Reply:
That would definitely work.
I was disliking all the operations when matching a regex could get the job done, but it boils down preference- it's up to guys how you want to do it...

Reply:
fixed in 3d3c8add5a6a1bb9ef396db5a088d07373068bbe